### PR TITLE
Use testcontainers for testing

### PR DESCRIPTION
**Pull request by hgeraldino**
I found it pretty handy to have the tests being self-contained (i.e. not requiring users to spin up docker separately)

### CODE REVIEW

File: src/test/scala/support/Mongo.scala [hunk 1]
```diff
@@ -27,7 +28,10 @@ object Mongo {
     DEFAULT_CODEC_REGISTRY
   )
 
-  lazy val mongoClient = MongoClient("mongodb://localhost:27017")
+  lazy val mongoDBContainer = new MongoDBContainer(DockerImageName.parse("mongo:4.0.10"))
```
Comment (marc0der) [inline src/test/scala/support/Mongo.scala]: I think we're still on mongo 3.2 atm. Let's do the upgrade as a separate piece of work as it affects many services.

File: src/test/scala/support/Mongo.scala [hunk 2]
```diff
@@ -27,7 +28,10 @@ object Mongo {
...
+
+  lazy val mongoClient = MongoClient("mongodb://%s:%d".format(
```
Comment (marc0der) [inline src/test/scala/support/Mongo.scala]: Probably better to use Scala's string interpolation here.

File: src/test/scala/support/Mongo.scala [hunk 3]
```diff
@@ -37,6 +41,19 @@ object Mongo {
 
   lazy val candidatesCollection: MongoCollection[Candidate] = db.getCollection("candidates")
 
+  def startMongoDb() = {
```
Comment (marc0der) [inline src/test/scala/support/Mongo.scala]: It's just stylistic, but these are simple expressions so they don't need curly braces. Same for all the following functions.

File: src/test/scala/io/sdkman/repos/ApplicationRepoSpec.scala [hunk 4]
```diff
@@ -40,11 +41,17 @@ class ApplicationRepoSpec
     }
   }
 
+  override def beforeAll() = {
```
Comment (marc0der) [inline src/test/scala/io/sdkman/repos/ApplicationRepoSpec.scala]: An expression, so can simply be something like this:
```suggestion
  override def beforeAll() = Mongo.startMongoDb()
```

File: src/test/scala/io/sdkman/repos/ApplicationRepoSpec.scala [hunk 5]
```diff
@@ -40,11 +41,17 @@ class ApplicationRepoSpec
...
+  }
+
   before {
     Mongo.dropAllCollections()
   }
 
   private trait TestRepo extends ApplicationRepo with MongoConnectivity with MongoConfiguration {
     override val config: Config = ConfigFactory.load()
+      .withValue("mongo.url.host",  ConfigValueFactory.fromAnyRef(Mongo.getMongoDbHost()))
+      .withValue("mongo.url.port",  ConfigValueFactory.fromAnyRef(Mongo.getMongoDbPort()))
```
Comment (marc0der) [inline src/test/scala/io/sdkman/repos/ApplicationRepoSpec.scala]:
```suggestion
  override val config: Config = ConfigFactory.parseString("""
      mongo.url.host=${Mongo.getMongoDbHost()}
      mongo.url.port=${Mongo.getMongoDbPort()}
      """.trimIndent()
  ).withFallback(ConfigFactory.load())

```

File: src/test/scala/support/Mongo.scala (hunk 1 above)
Comment (hgeraldino) [reply src/test/scala/support/Mongo.scala]: done

File: src/test/scala/support/Mongo.scala (hunk 2 above)
Comment (hgeraldino) [reply src/test/scala/support/Mongo.scala]: You can see my Java leaking here :)

File: src/test/scala/support/Mongo.scala (hunk 3 above)
Comment (hgeraldino) [reply src/test/scala/support/Mongo.scala]: And here as well

File: src/test/scala/io/sdkman/repos/ApplicationRepoSpec.scala (hunk 5 above)
Comment (hgeraldino) [reply src/test/scala/io/sdkman/repos/ApplicationRepoSpec.scala]: done

**Review by marc0der — APPROVED** [review]

**Pull request merged**